tests/unit/aws/protocol/test_parser.py

```diff
@@ -17,7 +17,7 @@
 )
 from localstack.aws.spec import load_service
 from localstack.http import Request as HttpRequest
-from localstack.services.s3 import s3_utils
+from localstack.services.s3.legacy import s3_utils
```

Comment:
Maybe we could - instead of importing the legacy module here - just copy the code here to avoid the import.
This import is only used for the `get_bucket_hostname`, which in turn is only used in this test. It also is just one single, trivial line:
```
"%s.%s:%s" % (bucket_name, S3_VIRTUAL_HOSTNAME, config.EDGE_PORT)
```

I think it would be awesome if we could get rid of this import now instead of stumbling across it when we remove the legacy module.
This also is - as far as I can see - the only legacy import in a non-legacy codepath. This means that we could just simply remove the legacy module, the providers, and some legacy codepaths (like in `generic_proxy` or the cloudformation `api_utils`) later on without any complications. 🚀

Reply:
This test only validates the behaviour of the legacy provider, so it should be removed as well when we delete the legacy module, that's why I explicitly left the import so that we could find it.
See the marker:
```python
@pytest.mark.skipif(
    not config.LEGACY_S3_PROVIDER, reason="ASF provider does not rely on virtual host parser"
)
```